src/Transport/ApiSnmpClient.php

```diff
@@ -89,32 +89,34 @@ public function getNext(array $oids) : array
     }
 
     /** @inheritDoc */
-    public function walk(string $oid) : array
+    public function walk(string $oid, int $maxRepetitions = 40) : array
     {
-        return $this->executeRequest(self::REQUEST_TYPE_WALK, [$oid]);
+        return $this->executeRequest(self::REQUEST_TYPE_WALK, [$oid], $maxRepetitions);
     }
 
     /**
      * @param string[] $oids
      *
      * @return array<string, mixed>
      */
-    private function executeRequest(string $requestType, array $oids) : array
+    private function executeRequest(string $requestType, array $oids, ?int $maxRepetitions = null) : array
```

Comment:
Koukám na mobilu. Ten null může nastat?

Reply:
je to private metoda, sdílená pro všechny operace. MaxRepetitions se používá jen pro walk, takže pro get/getNext to bude null